src/command/client/search/interactive.rs

```diff
@@ -288,48 +211,136 @@ impl State {
             )
             .split(chunks[0]);
 
-        let title = Paragraph::new(Text::from(Span::styled(
-            format!("Atuin v{VERSION}"),
-            Style::default().fg(Color::DarkGray),
-        )));
+        let title = self.build_title();
+        f.render_widget(title, header_chunks[0]);
+
+        let help = self.build_help();
+        f.render_widget(help, header_chunks[1]);
+
+        let stats = self.build_stats();
+        f.render_widget(stats, header_chunks[2]);
+
+        let results_list = Self::build_results_list(compact, results);
+        f.render_stateful_widget(results_list, chunks[1], &mut self.results_state);
 
+        let input = self.build_input(compact, chunks[2].width.into());
+        f.render_widget(input, chunks[2]);
+
+        let preview = self.build_preview(results, compact, preview_width, chunks[3].width.into());
+        f.render_widget(preview, chunks[3]);
+
+        let extra_width = UnicodeWidthStr::width(self.input.substring());
+
+        let cursor_offset = u16::from(!compact);
+        f.set_cursor(
+            // Put cursor past the end of the input text
+            chunks[2].x + extra_width as u16 + PREFIX_LENGTH + 1 + cursor_offset,
+            chunks[2].y + cursor_offset,
+        );
+    }
+
+    fn build_title(&mut self) -> Paragraph {
+        let title = if self.update_needed.is_some() {
+            let version = self.update_needed.clone().unwrap();
+
+            Paragraph::new(Text::from(Span::styled(
+                format!(" Atuin v{VERSION} - UPDATE AVAILABLE {version}"),
+                Style::default().add_modifier(Modifier::BOLD).fg(Color::Red),
+            )))
+        } else {
+            Paragraph::new(Text::from(Span::styled(
+                format!(" Atuin v{VERSION}"),
+                Style::default().add_modifier(Modifier::BOLD),
+            )))
+        };
+        title
+    }
+
+    #[allow(clippy::unused_self)]
```

Comment:
I preferred to add this than turning the function into an associated one to keep things uniform.